### PR TITLE
[CatalogPromotion][Behat] Reapplying multiple catalog promotions

### PR DESCRIPTION
**Pull request by AdamKasp**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| License         | MIT

### CODE REVIEW

File: features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    Scenario: Reapplying catalog promotion after adding new channel to them
+        When I make "Winter sale" catalog promotion available in the "Web-GB" channel
+        Then the visitor should see "$35.00" as the price of the "T-Shirt" product in the "Web-GB" channel
+        And the visitor should see "$100.00" as the original price of the "T-Shirt" product in the "Web-GB" channel
+        And the visitor should still see "$35.00" as the price of the "T-Shirt" product in the "Web-US" channel
+        And the visitor should still see "$100.00" as the original price of the "T-Shirt" product in the "Web-US" channel
+
+    @api @todo
+    Scenario: Reapplying catalog promotion after switching availability in channels
+        When I switch "Winter sale" catalog promotion availability from the "Web-US" channel to the "Web-GB" channel
```
Comment (lchrusciel) [inline features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature:34]:
```suggestion
        When I switch the "Winter sale" catalog promotion availability from the "Web-US" channel to the "Web-GB" channel
```

**Review by lchrusciel — APPROVED** [review]

File: features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        And the store has a "T-Shirt" configurable product
+        And this product has "PHP T-Shirt" variant priced at "$100.00" in "Web-US" channel
+        And "PHP T-Shirt" variant priced at "$30.00" in "Web-GB" channel
+        And this product is available in "Web-US" channel and "Web-GB" channel
+        And there is a catalog promotion "Winter sale" available in "Web-US" channel that reduces price by "30%" and applies on "PHP T-shirt" variant
+        And there is a another catalog promotion "Christmas sale" available in "Web-US" channel and "Web-GB" channel that reduces price by "50%" and applies on "PHP T-shirt" variant
+        And I am logged in as an administrator
+
+    @api @todo
+    Scenario: Removing only modified catalog promotion after removing its channel
```
Comment (GSadee) [inline features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature:19]: This scenario looks the same as the one from `features/promotion/applying_catalog_promotions/reapplying_catalog_promotions_after_editing_its_channels.feature:18`. 
I would be for modifying it to have variant with 2 catalog promotions applied and remove only one of them by changing channel or just remove a channel from `Web-GB` channel 🤔

File: features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        When I make "Winter sale" catalog promotion unavailable in the "Web-US" channel
+        Then the visitor should see "$50.00" as the price of the "T-Shirt" product in the "Web-US" channel
+        And the visitor should see "$100.00" as the original price of the "T-Shirt" product in the "Web-US" channel
+
+    @api @todo
+    Scenario: Reapplying catalog promotion after adding new channel to them
+        When I make "Winter sale" catalog promotion available in the "Web-GB" channel
+        Then the visitor should see "$35.00" as the price of the "T-Shirt" product in the "Web-GB" channel
+        And the visitor should see "$100.00" as the original price of the "T-Shirt" product in the "Web-GB" channel
```
Comment (GSadee) [inline features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature:28]: The prices in `Web-GB` channel should be different than in `Web-US`

File: features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        And the visitor should still see "$35.00" as the price of the "T-Shirt" product in the "Web-US" channel
+        And the visitor should still see "$100.00" as the original price of the "T-Shirt" product in the "Web-US" channel
+
+    @api @todo
+    Scenario: Reapplying catalog promotion after switching availability in channels
```
Comment (GSadee) [inline features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature:33]: This scenario looks the same as the one from `features/promotion/applying_catalog_promotions/reapplying_catalog_promotions_after_editing_its_channels.feature:32`
I would be for modifying this scenario to have a store with 3 channels, a variant with 2 catalog promotions applied and one them will be modified to change one of its channel

File: features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature (hunk 2 above)
Comment (GSadee) [reply features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature:19]: I'm resolving as it was an invalid comment 🎉

File: features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature (hunk 4 above)
Comment (GSadee) [reply features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature:33]: I'm resolving as it was an invalid comment 🎉

File: features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature (hunk 2 above)
Comment (AdamKasp) [reply features/promotion/applying_catalog_promotions/reapplying_multiple_catalog_promotions_after_editing_their_channels.feature:19]: they are similar but not the same, the main difference is in line 15 (where I've configured second CP)

**Comment by AdamKasp** [comment]
replaced by #13124

**Pull request closed without merging**